hastory-server/src/Data/Hastory/Server/Handler/Sessions.hs

```diff
@@ -0,0 +1,21 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Data.Hastory.Server.Handler.Sessions where
+
+import Data.Hastory.Server.Handler.Import
+
+createSessionHandler :: ServerSettings -> UserForm -> Handler (Headers AuthCookies NoContent)
+createSessionHandler ServerSettings {..} UserForm {..} = do
+  user <-
+    entityVal <$> (runDB (getBy $ UniqueUsername userFormUserName) _ssDbPool >>= ensureWith err401)
+  let passwordAccepted =
+        checkPassword userFormPassword (userHashedPassword user) == PasswordCheckSuccess
```

Comment:
instead of building a boolean, you can just pattern-match on the `PasswordCheckResult`.